unblob/handlers/archive/zip.py

```diff
@@ -147,7 +155,21 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
             file.seek(offset, io.SEEK_SET)
             end_of_central_directory = self.parse_header(file)
 
-            if self.is_zip64_eocd(end_of_central_directory):
+            # some values in the CD can be FFFF, indicating its a zip64
+            # if the offset of the CD is 0xFFFFFFFF, its definitely one
+            # otherwise we check every other header indicating zip64
+            if end_of_central_directory.offset_of_cd == 0xFFFFFFFF:
+                file.seek(offset, io.SEEK_SET)
+                end_of_central_directory = self._parse_zip64(file, start_offset, offset)
+                break
+
+            file.seek(end_of_central_directory.offset_of_cd, io.SEEK_SET)
```

Comment:
If the ZIP file is prefixed by garbage data, you'll seek to the wrong place.

```suggestion
            file.seek(start_offset + end_of_central_directory.offset_of_cd, io.SEEK_SET)
```

Reply:
I'd add a sample with some prefixed data to test that.

Reply:
Modified the code with the `start_offset`, I will add the integration files

Reply:
Added integration tests using dd with /dev/random to have 1024 bytes of random data in front of the file.